File: packages/android/src/main/java/com/kozmos/components/Stepper/Stepper.figma.kt
package com.kozmos.components.stepper

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=512-39614")
class KozmosStepperConnect {
    @FigmaProperty(FigmaType.Enum, "Count")
    val count: Int = Figma.mapping(
        "2" to 2,
        "3" to 3,
        "4" to 4
    )

    @FigmaProperty(FigmaType.Enum, "Current")
    val currentStep: Int = Figma.mapping(
        "1" to 0,
        "2" to 1,
        "3" to 2,
        "4" to 3
    )

    @FigmaProperty(FigmaType.Text, "Step 1 Text")
    val step1Text: String = "Start"

    @FigmaProperty(FigmaType.Text, "Step 2 Text")
    val step2Text: String = "Review"

    @FigmaProperty(FigmaType.Text, "Step 3 Text")
    val step3Text: String = "Confirm"

    @FigmaProperty(FigmaType.Text, "Step 4 Text")
    val step4Text: String = "Done"

    @Composable
    fun ComponentExample() {
        KozmosStepper(
            steps = listOf(step1Text, step2Text, step3Text, step4Text).take(count),
            currentStep = currentStep.coerceAtMost(count - 1)
        )
    }
}

File: packages/ios/Sources/Components/Stepper/Stepper.figma.swift
import SwiftUI
import Figma

struct KozmosStepperConnect: FigmaConnect {
    let component = KozmosStepper.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=512-39614"

    @FigmaEnum(
        "Count",
        mapping: [
            "2": 2,
            "3": 3,
            "4": 4
        ]
    )
    var count: Int = 4

    @FigmaEnum(
        "Current",
        mapping: [
            "1": 0,
            "2": 1,
            "3": 2,
            "4": 3
        ]
    )
    var currentStep: Int = 0

    @FigmaString("Step 1 Text")
    var step1Text: String = "Start"

    @FigmaString("Step 2 Text")
    var step2Text: String = "Review"

    @FigmaString("Step 3 Text")
    var step3Text: String = "Confirm"

    @FigmaString("Step 4 Text")
    var step4Text: String = "Done"

    private var stepLabels: [String] {
        Array([self.step1Text, self.step2Text, self.step3Text, self.step4Text].prefix(self.count))
    }

    var body: some View {
        KozmosStepper(
            steps: self.stepLabels,
            currentStep: min(self.currentStep, self.count - 1)
        )
    }
}
